src/controllers/weather_controller.py

```diff
@@ -1,37 +1,74 @@
 import requests
 from datetime import datetime, timedelta
+import os
+import csv
+from io import StringIO
 
 class WeatherController:
     HTTP_STATUS_OK = 200
+    DATE_LENGTH = 10
+    SUNNY_IRRADIANCE_THRESHOLD = 200
 
     def __init__(self, api_key):
         self.api_key = api_key
 
     def get_historical_weather(self, start_date, end_date, lat, lon):
-        base_url = "https://api.weatherbit.io/v2.0/history/daily"
-
+        base_url = "https://api.weather.com/v3/wx/hod/r1/direct"
```

Comment:
I'd recommend moving `base_url = "https://api.weather.com/v3` definition to the class variable and keeping the end-point in the individual functions. This is done to future-proof for example if in the future `v4` of the weather API is released it would easier to change. 
```
class WeatherController:
   ...
   BASE_URL= "https://api.weather.com/v3"
   ...

   def get_historical_weather():
      api_url = BASE_URL + "/wx/hod/r1/direct"
```
```